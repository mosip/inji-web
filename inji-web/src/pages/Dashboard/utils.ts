export const convertStringIntoPascalCase = (text: string | null) => {
    if (text != null) {
        return text
            .toLocaleLowerCase()
            .split(" ")
            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
            .join(" ");
    }
};
