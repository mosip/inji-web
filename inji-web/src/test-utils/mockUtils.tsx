import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

// Mock for storage module
export const mockStorageModule = () => {
    jest.mock('../utils/storage.ts', () => ({
        storage: {
            getItem: jest.fn(),
            setItem: jest.fn(),
            SESSION_INFO: 'SESSION_INFO',
            SELECTED_LANGUAGE: 'selectedLanguage'
        }
    }));
};

// Mock for localStorage
export const mockLocalStorage = () => {
    let store: { [key: string]: string } = {};
  
    const localStorageMock = {
      getItem: jest.fn((key: string) => store[key] || null),
      setItem: jest.fn((key: string, value: string) => {
        store[key] = value;
      })
    };
  
    Object.defineProperty(window, 'localStorage', { 
      value: localStorageMock,
      writable: true
    });
  
    return localStorageMock;
  };

// API Mock
export const mockApi = {
    authorizationRedirectionUrl: 'http://mockurl.com'
};

// Crypto Mock
export const mockCrypto = {
    getRandomValues: (array: Uint32Array) => {
        for (let i = 0; i < array.length; i++) {
            array[i] = 4294967296;
        }
        return array;
    },
    subtle: {} as SubtleCrypto,
    randomUUID: () => '123e4567-e89b-12d3-a456-426614174000'
} as Crypto;

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
 };

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};
              
export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};
    
export const mockUseParams = ()=>{
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({ issuerId: 'test-issuer-id' }),
      }));   
};

export const mockUseapiObject = () =>{
    jest.mock('../utils/api', () => ({
        api: {
          mimotoHost: 'http://mocked-api-host',
        },
      }));
};

export const mockUseFetch = () =>{
    jest.mock('../hooks/useFetch', () => {
        const RequestStatus = {
          LOADING: 'LOADING',
          DONE: 'DONE',
          ERROR: 'ERROR',
        };
        return {
          useFetch: () => jest.fn(),
          RequestStatus,
        };
      });
};

export const mockUseToast = () =>{
    jest.mock('react-toastify', () => ({
      toast: {
        error: jest.fn(),
      },
    }));
};

export const mockusemisc = ()=>{
    jest.mock('../utils/misc', () => ({
        downloadCredentialPDF: jest.fn(),
        getErrorObject: jest.fn(),
        getTokenRequestBody: jest.fn(),
      }));
};

export const mockusei18n = ()=>{
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key: string) => key,
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

};

export const mockWindowLocation = (url: string) => {
    const location = new URL(url) as unknown as Location;
    Object.defineProperty(window, 'location', {
      value: location,
      writable: true,
    });
  };

export const renderWithRouter = (Element: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
      <BrowserRouter>
        <Provider store={reduxStore}>
          <Routes>
            <Route path="*" element={Element} />
          </Routes>
        </Provider>
      </BrowserRouter>
    );
  };

