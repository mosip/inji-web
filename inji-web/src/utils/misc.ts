import sha256 from 'crypto-js/sha256';
import Base64 from 'crypto-js/enc-base64';
import {api} from "./api";

export const generateCodeChallenge = (verifier = generateRandomString()) => {
    const hashedVerifier = sha256(verifier);
    const base64Verifier = Base64.stringify(hashedVerifier);
    return {
        codeChallenge: base64Verifier
            .replace(/=/g, '')
            .replace(/\+/g, '-')
            .replace(/\//g, '_'),
        codeVerifier: verifier
    };
}

export const generateRandomString = (length = 43, charset='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~') => {
    let randomString = '';
    for (let i = 0; i < 43; i++) {
        const array = new Uint32Array(1);
        const randomOffset = crypto.getRandomValues(array)[0] / (2 ** 32) * charset.length;
        const randomIndex = Math.floor( randomOffset );
        randomString += charset[randomIndex];
    }
    return randomString;
};

export const isObjectEmpty = (object: any) => {
    return object === null || object === undefined || Object.keys(object).length === 0;
}

export const getTokenRequestBody = (code: string, codeVerifier: string, issuerId: string, credentialType: string, vcStorageExpiryLimitInTimes: string, locale: string) => {
    return {
        'grant_type': 'authorization_code',
        'code': code,
        'redirect_uri': api.authorizationRedirectionUrl,
        'code_verifier': codeVerifier,
        'issuer': issuerId,
        'credential': credentialType,
        'vcStorageExpiryLimitInTimes': vcStorageExpiryLimitInTimes,
        'locale':locale
    }
}

export const downloadCredentialPDF = async (response: any, certificateId: string) => {
    let fileName = `${certificateId}.pdf`;
    const url = window.URL.createObjectURL(response);
    const link = document.createElement('a');
    link.href = url;
    link.setAttribute('download', fileName);
    link.setAttribute('target', '_blank');
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    window.URL.revokeObjectURL(url);
}

export const getErrorObject = (downloadResponse: any) => {
    const errorCode = downloadResponse?.errors ? downloadResponse?.errors[0]?.errorCode : "";
    if([
        "err_missing_issuancedate",
        "err_invalid_issuancedate",
        "err_issuance_date_is_future_date",
        "err_invalid_expirationdate",
        "err_vc_expired",
        "err_invalid_validfrom",
        "err_valid_from_is_future_date",
        "err_invalid_validuntil"
    ].indexOf(errorCode) != -1 ){
        return {
            code: `error.verification.${errorCode}.title`,
            message: `error.verification.${errorCode}.subTitle`
        }
    }
    return {
        code: "error.generic.title",
        message: "error.generic.subTitle"
    }
}
export const constructContent = (descriptions: string[],applyHTML:boolean) => {
    return descriptions.map((desc, index) => {
        if (applyHTML) {
            return { __html: desc };
        }
        return desc;
    });
};
