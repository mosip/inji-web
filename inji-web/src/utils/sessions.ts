import {AppStorage} from "./AppStorage";
import {SessionObject} from "../types/data";

export const addNewSession = (session: SessionObject) => {
    let activeSessions = getAllActiveSession();
    activeSessions = activeSessions.concat([session]);
    AppStorage.setItem(AppStorage.SESSION_INFO, JSON.stringify(activeSessions));
}

export const getAllActiveSession: () => SessionObject[] = () => {
    const activeSessions = AppStorage.getItem(AppStorage.SESSION_INFO);
    if (activeSessions) {
        return JSON.parse(activeSessions);
    }
    return [];
}

export const getActiveSession = (state: string | null) => {
    const activeSession = getAllActiveSession().filter(session => session.state === state);
    return activeSession.length > 0 ? activeSession[0] : {};
}

export const removeActiveSession = (state: string) => {
    const remainingSessions = getAllActiveSession().filter(session => session.state !== state);
    AppStorage.setItem(AppStorage.SESSION_INFO, JSON.stringify(remainingSessions));
}

