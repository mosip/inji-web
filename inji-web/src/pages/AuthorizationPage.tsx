import React, {useEffect, useState} from "react";
import {SpinningLoader} from "../components/Common/SpinningLoader";
import {api} from "../utils/api";
import {LandingPageWrapper} from "../components/Common/LandingPageWrapper";
import {ErrorSheildIcon} from "../components/Common/ErrorSheildIcon";
import {useTranslation} from "react-i18next";

export const AuthorizationPage: React.FC = () => {
    const url = window.location.href;
    const currentQueryParams = new URLSearchParams(window.location.search);
    const [error, setError] = useState<String>(currentQueryParams.get("error")+"");
    const {t} = useTranslation("AuthorizationPage");

    useEffect( () => {
        if(url.indexOf("error") === -1) {
            window.location.href = api.mimotoHost + "/authorize" + window.location.search;
        }
    },[])

    if(url.indexOf("error") == -1){
        return <div><LandingPageWrapper icon={<SpinningLoader/>} title={""} subTitle={""} gotoHome={false}/></div>
    }
    return <div><LandingPageWrapper icon={<ErrorSheildIcon />} title={t(`error.code.${error}`)} subTitle={t(`error.message.${error}`)} gotoHome={true}/></div>
}


