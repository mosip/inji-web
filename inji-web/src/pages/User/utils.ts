import { User } from "../../components/Dashboard/types";
import { KEYS } from "../../utils/constants";
import {ROUTES} from "../../constants/Routes";

export const convertStringIntoPascalCase = (text: string | undefined) => {
    return (
        text?.toLocaleLowerCase()
            .split(' ')
            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
            .join(' ')
    );
};

export const navigateToUserHome = (navigate: any) =>
    navigate(ROUTES.USER_HOME);

export const validateWalletUnlockStatus = (
    cachedWalletId: string | null,
    storageWalletId: string | null,
    navigate: (path: string) => void,
    user: User
) => {
    if (cachedWalletId && (cachedWalletId === storageWalletId)) {
        console.info('Wallet is unlocked!');
    } else {
        console.warn(
            'Wallet exists but is locked, redirecting to `/pin` to unlock the wallet.'
        );
        if(user) {
            navigate('/pin');
            localStorage.removeItem(KEYS.WALLET_ID);
        }
    }
};