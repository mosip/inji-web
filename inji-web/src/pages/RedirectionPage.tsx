import React, {useEffect, useState} from 'react';
import {getActiveSession, removeActiveSession} from "../utils/sessions";
import {useLocation} from "react-router-dom";
import {NavBar} from "../components/Common/NavBar";
import {RequestStatus, useFetch} from "../hooks/useFetch";
import {DownloadResult} from "../components/Redirection/DownloadResult";
import {api} from "../utils/api";
import {ApiRequest, DisplayArrayObject, SessionObject} from "../types/data";
import {useTranslation} from "react-i18next";
import {downloadCredentialPDF} from "../utils/misc";
import {getObjectForCurrentLanguage} from "../utils/i18n";

export const RedirectionPage: React.FC = () => {

    const {error, state, fetchRequest} = useFetch();
    const location = useLocation();
    const searchParams = new URLSearchParams(location.search);
    const redirectedSessionId = searchParams.get("state");
    const activeSessionInfo: any = getActiveSession(redirectedSessionId);
    const {t} = useTranslation("RedirectionPage");
    const [session, setSession] = useState<SessionObject | null>(activeSessionInfo);
    const displayObject = getObjectForCurrentLanguage(session?.selectedIssuer?.display ?? []);

    useEffect(() => {
        const fetchToken = async () => {
            if (Object.keys(activeSessionInfo).length > 0) {
                const code = searchParams.get("code") ?? "";
                const urlState = searchParams.get("state") ?? "";
                const clientId = activeSessionInfo?.selectedIssuer.client_id;
                const codeVerifier = activeSessionInfo?.codeVerifier;
                const issuerId = activeSessionInfo?.selectedIssuer.credential_issuer ?? "";
                const certificateId = activeSessionInfo?.certificateId;

                const bodyJson = {
                    'grant_type': 'authorization_code',
                    'code': code,
                    'client_id': clientId,
                    'client_assertion_type': 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
                    'client_assertion': '',
                    'redirect_uri': api.authorizationRedirectionUrl,
                    'code_verifier': codeVerifier
                }
                const requestBody = new URLSearchParams(bodyJson);

                let apiRequest: ApiRequest = api.fetchToken;
                let response = await fetchRequest(
                    apiRequest.url(issuerId),
                    apiRequest.methodType,
                    apiRequest.headers(),
                    requestBody
                );

                apiRequest = api.downloadVc;
                response = await fetchRequest(
                    apiRequest.url(issuerId, certificateId),
                    apiRequest.methodType,
                    apiRequest.headers(response?.access_token)
                );
                if (state !== RequestStatus.ERROR) {
                    await downloadCredentialPDF(response, certificateId);
                }
                if (urlState != null) {
                    removeActiveSession(urlState);
                }
            } else {
                setSession(null);
            }
        }
        fetchToken();

    }, [])

    const loadStatusOfRedirection = () => {
        if (!session) {
            return <DownloadResult title={t("error.invalidSession.title")}
                                   subTitle={t("error.invalidSession.subTitle")}
                                   state={RequestStatus.ERROR}/>
        }
        if (state === RequestStatus.LOADING) {
            return <DownloadResult title={t("loading.title")}
                            subTitle={t("loading.subTitle")}
                            state={RequestStatus.LOADING}/>
        }
        if (state === RequestStatus.ERROR && error) {
            return <DownloadResult title={t("error.generic.title")}
                                   subTitle={t("error.generic.subTitle")}
                                   state={RequestStatus.ERROR}/>
        }
        return <DownloadResult title={t("success.title")}
                               subTitle={t("success.subTitle")}
                               state={RequestStatus.DONE}/>
    }

    return <div data-testid="Redirection-Page-Container">
        {activeSessionInfo?.selectedIssuer?.credential_issuer && <NavBar title={displayObject?.name ?? ""} search={false} link={`/issuers/${activeSessionInfo?.selectedIssuer?.credential_issuer}`}/>}
        {loadStatusOfRedirection()}
    </div>
}
