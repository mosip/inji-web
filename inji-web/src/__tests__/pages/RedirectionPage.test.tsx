import React from 'react';
import { RedirectionPage } from '../../pages/RedirectionPage';
import { getActiveSession} from '../../utils/sessions';
import { downloadCredentialPDF, getErrorObject } from '../../utils/misc';
import { renderWithRouter,mockusei18n,mockUseFetch} from '../../test-utils/mockUtils';


//todo : extract the local method to mockUtils, which is added to bypass the routing problems
// Mock the utility functions
jest.mock('../../utils/sessions', () => ({
  getActiveSession: jest.fn(),
  removeActiveSession: jest.fn(),
}));
jest.mock('../../utils/misc', () => ({
  downloadCredentialPDF: jest.fn(),
  getErrorObject: jest.fn(),
  getTokenRequestBody: jest.fn(),
}));
const mockUseFetchhook = jest.fn();
describe('Testing the Layout of RedirectionPage', () => {
  mockusei18n();
  mockUseFetch();
  test('Check if the layout is matching with the snapshots', () => {
    mockUseFetchhook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    (getActiveSession as jest.Mock).mockReturnValue({ selectedIssuer: { issuer_id: 'issuer1', display: [{ name: 'Test Issuer' }] } });
    const { asFragment } = renderWithRouter(<RedirectionPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Testing the Functionality of RedirectionPage', () => {
  beforeEach(() => {
    mockusei18n();
    mockUseFetch();
    jest.clearAllMocks();
    (getActiveSession as jest.Mock).mockReturnValue({ selectedIssuer: { issuer_id: 'issuer1', display: [{ name: 'Test Issuer' }] } });
  });

  test('Check if NavBar component is rendered', () => {
    mockUseFetchhook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    const{asFragment} =  renderWithRouter(<RedirectionPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Check if it displays error message if state is ERROR', async () => {
    mockUseFetchhook.mockReturnValue({ state: 'ERROR', fetchRequest: jest.fn(), error: true, response: {} });
    (getErrorObject as jest.Mock).mockReturnValue({ code: 'error.generic.title', message: 'error.generic.subTitle' });
    const{asFragment} =  renderWithRouter(<RedirectionPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Check if DownloadResult component shows loading state', () => {
    mockUseFetchhook.mockReturnValue({ state: 'LOADING', fetchRequest: jest.fn() });
    const{asFragment} = renderWithRouter(<RedirectionPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Check if DownloadResult component shows success state', async () => {
    mockUseFetchhook.mockReturnValue({ state: 'DONE', fetchRequest: jest.fn() });
    (downloadCredentialPDF as jest.Mock).mockResolvedValueOnce(true);
    const{asFragment} = renderWithRouter(<RedirectionPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
