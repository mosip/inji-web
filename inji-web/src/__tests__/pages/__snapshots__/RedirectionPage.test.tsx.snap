// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Functionality of RedirectionPage Check if DownloadResult component shows loading state 1`] = `
<DocumentFragment>
  <div
    data-testid="Redirection-Page-Container"
  >
    <div
      class="bg-gradient-to-r from-iw-lightPrimary to-iw-lightSecondary text-iw-title p-4 py-10"
      data-testid="NavBar-Outer-Container"
    >
      <nav
        class=" mx-auto flex flex-col justify-start container items-start sm:flex-row sm:justify-start sm:items-center"
        data-testid="NavBar-Inner-Container"
      >
        <div
          class="flex items-center my-5 sm:my-0"
        >
          <div
            class="cursor-pointer"
          >
            <svg
              data-testid="NavBar-Back-Arrow"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244 400 100 256l144-144M120 256h292"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="48"
              />
            </svg>
          </div>
          <span
            class="text-2xl font-semibold ps-2 pe-4 md:whitespace-nowrap whitespace-wrap"
            data-testid="NavBar-Text"
          />
        </div>
      </nav>
    </div>
    <div
      class="flex flex-col justify-center items-center pt-32 "
      data-testid="DownloadResult-Outer-Container"
    >
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="my-4"
      >
        <p
          class="font-bold"
          data-testid="DownloadResult-Title"
        >
          Download in progress...
        </p>
      </div>
      <div
        class="mb-6 px-10 text-center"
        data-testid="DownloadResult-SubTitle"
      >
        <p>
          Please wait a moment! We're converting your verifiable credentials to PDF
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Functionality of RedirectionPage Check if DownloadResult component shows success state 1`] = `
<DocumentFragment>
  <div
    data-testid="Redirection-Page-Container"
  >
    <div
      class="bg-gradient-to-r from-iw-lightPrimary to-iw-lightSecondary text-iw-title p-4 py-10"
      data-testid="NavBar-Outer-Container"
    >
      <nav
        class=" mx-auto flex flex-col justify-start container items-start sm:flex-row sm:justify-start sm:items-center"
        data-testid="NavBar-Inner-Container"
      >
        <div
          class="flex items-center my-5 sm:my-0"
        >
          <div
            class="cursor-pointer"
          >
            <svg
              data-testid="NavBar-Back-Arrow"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244 400 100 256l144-144M120 256h292"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="48"
              />
            </svg>
          </div>
          <span
            class="text-2xl font-semibold ps-2 pe-4 md:whitespace-nowrap whitespace-wrap"
            data-testid="NavBar-Text"
          />
        </div>
      </nav>
    </div>
    <div
      class="flex flex-col justify-center items-center pt-32 "
      data-testid="DownloadResult-Outer-Container"
    >
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="my-4"
      >
        <p
          class="font-bold"
          data-testid="DownloadResult-Title"
        >
          Download in progress...
        </p>
      </div>
      <div
        class="mb-6 px-10 text-center"
        data-testid="DownloadResult-SubTitle"
      >
        <p>
          Please wait a moment! We're converting your verifiable credentials to PDF
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Functionality of RedirectionPage Check if NavBar component is rendered 1`] = `
<DocumentFragment>
  <div
    data-testid="Redirection-Page-Container"
  >
    <div
      class="bg-gradient-to-r from-iw-lightPrimary to-iw-lightSecondary text-iw-title p-4 py-10"
      data-testid="NavBar-Outer-Container"
    >
      <nav
        class=" mx-auto flex flex-col justify-start container items-start sm:flex-row sm:justify-start sm:items-center"
        data-testid="NavBar-Inner-Container"
      >
        <div
          class="flex items-center my-5 sm:my-0"
        >
          <div
            class="cursor-pointer"
          >
            <svg
              data-testid="NavBar-Back-Arrow"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244 400 100 256l144-144M120 256h292"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="48"
              />
            </svg>
          </div>
          <span
            class="text-2xl font-semibold ps-2 pe-4 md:whitespace-nowrap whitespace-wrap"
            data-testid="NavBar-Text"
          />
        </div>
      </nav>
    </div>
    <div
      class="flex flex-col justify-center items-center pt-32 "
      data-testid="DownloadResult-Outer-Container"
    >
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="my-4"
      >
        <p
          class="font-bold"
          data-testid="DownloadResult-Title"
        >
          Download in progress...
        </p>
      </div>
      <div
        class="mb-6 px-10 text-center"
        data-testid="DownloadResult-SubTitle"
      >
        <p>
          Please wait a moment! We're converting your verifiable credentials to PDF
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Functionality of RedirectionPage Check if it displays error message if state is ERROR 1`] = `
<DocumentFragment>
  <div
    data-testid="Redirection-Page-Container"
  >
    <div
      class="bg-gradient-to-r from-iw-lightPrimary to-iw-lightSecondary text-iw-title p-4 py-10"
      data-testid="NavBar-Outer-Container"
    >
      <nav
        class=" mx-auto flex flex-col justify-start container items-start sm:flex-row sm:justify-start sm:items-center"
        data-testid="NavBar-Inner-Container"
      >
        <div
          class="flex items-center my-5 sm:my-0"
        >
          <div
            class="cursor-pointer"
          >
            <svg
              data-testid="NavBar-Back-Arrow"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244 400 100 256l144-144M120 256h292"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="48"
              />
            </svg>
          </div>
          <span
            class="text-2xl font-semibold ps-2 pe-4 md:whitespace-nowrap whitespace-wrap"
            data-testid="NavBar-Text"
          />
        </div>
      </nav>
    </div>
    <div
      class="flex flex-col justify-center items-center pt-32 "
      data-testid="DownloadResult-Outer-Container"
    >
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="my-4"
      >
        <p
          class="font-bold"
          data-testid="DownloadResult-Title"
        >
          Download in progress...
        </p>
      </div>
      <div
        class="mb-6 px-10 text-center"
        data-testid="DownloadResult-SubTitle"
      >
        <p>
          Please wait a moment! We're converting your verifiable credentials to PDF
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of RedirectionPage Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    data-testid="Redirection-Page-Container"
  >
    <div
      class="bg-gradient-to-r from-iw-lightPrimary to-iw-lightSecondary text-iw-title p-4 py-10"
      data-testid="NavBar-Outer-Container"
    >
      <nav
        class=" mx-auto flex flex-col justify-start container items-start sm:flex-row sm:justify-start sm:items-center"
        data-testid="NavBar-Inner-Container"
      >
        <div
          class="flex items-center my-5 sm:my-0"
        >
          <div
            class="cursor-pointer"
          >
            <svg
              data-testid="NavBar-Back-Arrow"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244 400 100 256l144-144M120 256h292"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="48"
              />
            </svg>
          </div>
          <span
            class="text-2xl font-semibold ps-2 pe-4 md:whitespace-nowrap whitespace-wrap"
            data-testid="NavBar-Text"
          />
        </div>
      </nav>
    </div>
    <div
      class="flex flex-col justify-center items-center pt-32 "
      data-testid="DownloadResult-Outer-Container"
    >
      <div
        class="flex justify-center items-center"
        data-testid="SpinningLoader-Container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          class="sc-beySPh gZmPOF"
          data-testid="oval-loading"
          role="progressbar"
        >
          <svg
            data-testid="oval-svg"
            height="80"
            stroke="var(--iw-color-spinningLoaderPrimary)"
            viewBox="-20.5 -20.5 43 43"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="3"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="var(--iw-color-spinningLoaderSecondary)"
                  stroke-opacity=".5"
                  stroke-width="3"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="my-4"
      >
        <p
          class="font-bold"
          data-testid="DownloadResult-Title"
        >
          Download in progress...
        </p>
      </div>
      <div
        class="mb-6 px-10 text-center"
        data-testid="DownloadResult-SubTitle"
      >
        <p>
          Please wait a moment! We're converting your verifiable credentials to PDF
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
